teps/0132-queueing-concurrent-pipelineruns.md

```diff
@@ -0,0 +1,313 @@
+---
+status: proposed
+title: Queueing Concurrent PipelineRuns
+creation-date: '2023-02-24'
+last-updated: '2023-02-24'
+authors:
+- '@lbernick'
+collaborators:
+- '@chengjoey'
+- '@jerop'
+---
+
+# TEP-0132: Queueing Concurrent PipelineRuns
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases](#use-cases)
+  - [Existing Workarounds](#existing-workarounds)
+  - [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Notes and Caveats](#notes-and-caveats)
+- [Design Details](#design-details)
+- [Design Evaluation](#design-evaluation)
+  - [Reusability](#reusability)
+  - [Simplicity](#simplicity)
+  - [Flexibility](#flexibility)
+  - [User Experience](#user-experience)
+  - [Performance](#performance)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Implementation Plan](#implementation-plan)
+  - [Test Plan](#test-plan)
+  - [Infrastructure Needed](#infrastructure-needed)
+  - [Upgrade and Migration Strategy](#upgrade-and-migration-strategy)
+  - [Implementation Pull Requests](#implementation-pull-requests)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+
+This TEP proposes allowing users to control the number of PipelineRuns that can run concurrently.
```

Comment:
why pipelineruns only? can we add taskruns and customruns in scope?

Reply:
I changed the title to "queueing concurrent runs" and updated the summary to include TaskRuns. Thanks for pointing this out! Do you know of any use cases for queueing Custom Runs? (I guess if they are matrixed, but it probably depends what the custom run does)

Reply:
agreed that it's mainly for matrix, we want to give users the option to queue customruns but it's up to them to decide whether it makes sense for their custom task

Reply:
thanks for updating the problem statement -- please update the PR title as well

wondering about TEP-0120 also being focused on PipelineRuns, should it apply to other Runs as well?

Reply:
Done, thanks. For TEP-0120, it's less clear that there's a strong use case for other types of runs so I kept it scoped to just PipelineRuns to reduce complexity in an initial version. We can definitely expand to other types of runs if there's a clear use case.